tests/components/template/test_sensor.py

```diff
@@ -1159,23 +1003,23 @@ async def test_trigger_not_allowed_platform_config(hass, caplog):
                 },
             }
         },
-    )
-
-    await hass.async_block_till_done()
-
-    state = hass.states.get("sensor.test_template_sensor")
+    ],
+)
+async def test_trigger_not_allowed_platform_config(hass, caplog, start_ha):
+    """Test we throw a helpful warning if a trigger is configured in platform config."""
+    state = hass.states.get(TEST_NAME)
     assert state is None
+    messages = str([x.message for x in caplog.get_records("setup")])
```

Comment:
Why do you prefer this over `caplog.text` ?

Reply:
Because the text I search is from the “setup” phase caplog.text only contains the current phase.

Took me a while to get this right. Caplog divides the entries in 3 separate parts (setup, run and teardown), actually quite confusing.

Reply:
Ah right, we're now setting it up in a fixture.